src/vector_v4.hpp

```diff
@@ -109,6 +109,7 @@ template <typename T> class VectorV4 {
 
     // size_
     int size() { return size_; }
+
```

Comment:
Resize is wrong.

You mentioned "delete[] temp;" is wrong. Fix it then ?

Did you test out resize with T = that struct with assert in copy-ctr, cpy-assignment ? The one we used Optional implementation ?
It will crash.